cap/worker.go

```diff
@@ -0,0 +1,82 @@
+package cap
+
+import (
+	"context"
+
+	"github.com/capatazlib/go-capataz/internal/c"
+)
+
+// NotifyStartFn is a function given to worker nodes that allows them to notify
+// the parent supervisor that they are officialy started.
+//
+// You may notify a worker fail to start by passing an error argument to this
```

Comment:
do you mean:
> You may notify the parent supervisor that a worker failed to start ...

Better yet:
> The argument contains an error, if there was a failure, or nil, otherwise.